### PR TITLE
storage/testing: a simple refactor

### PR DESCRIPTION
**Pull request by p0lyn0mial**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
A simple refactor (no-op) of the storage/testing package for https://github.com/kubernetes/kubernetes/pull/119557 

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/storage/testing/utils.go [hunk 1]
```diff
@@ -100,15 +100,22 @@ func testPropagateStore(ctx context.Context, t *testing.T, store storage.Interfa
...
-			t.Errorf("%s: %s", msg, diff)
-		} else {
-			t.Errorf("%s:\nexpected: %#v\ngot: %#v", msg, expected, got)
+	equal, diff := areObjectDeepEqual(expected, got)
+	if !equal {
+		t.Fatalf("%s: %s", msg, diff)
+	}
+}
+
+func areObjectDeepEqual(expected, actual interface{}) (bool, string) {
```
Comment (p0lyn0mial) [inline staging/src/k8s.io/apiserver/pkg/storage/testing/utils.go:111]: i'm going to use this function in my upcoming PR

File: staging/src/k8s.io/apiserver/pkg/storage/testing/utils.go [hunk 2]
```diff
@@ -100,13 +100,13 @@ func testPropagateStore(ctx context.Context, t *testing.T, store storage.Interfa
...
 
-func ExpectNoDiff(t *testing.T, msg string, expected, got interface{}) {
+func expectObjectsDeepEqual(t *testing.T, msg string, expected, actual interface{}) {
 	t.Helper()
-	if !reflect.DeepEqual(expected, got) {
-		if diff := cmp.Diff(expected, got); diff != "" {
-			t.Errorf("%s: %s", msg, diff)
+	if !reflect.DeepEqual(expected, actual) {
+		if diff := cmp.Diff(expected, actual); diff != "" {
+			t.Fatalf("%s: %s", msg, diff)
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/storage/testing/utils.go:107]: please revert - I actually prefer t.Errorf, as in case of multiple events we may get more detailed results.

File: staging/src/k8s.io/apiserver/pkg/storage/testing/utils.go [hunk 3]
```diff
@@ -100,13 +100,13 @@ func testPropagateStore(ctx context.Context, t *testing.T, store storage.Interfa
 	return key, setOutput
 }
 
-func ExpectNoDiff(t *testing.T, msg string, expected, got interface{}) {
+func expectObjectsDeepEqual(t *testing.T, msg string, expected, actual interface{}) {
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/storage/testing/utils.go:103]: I'm not entirely sure why did you rename it - can you please revert?

File: staging/src/k8s.io/apiserver/pkg/storage/testing/watcher_tests.go [hunk 4]
```diff
@@ -546,24 +546,24 @@ func RunOptionalTestProgressNotify(ctx context.Context, t *testing.T, store stor
...
 	// object of the correct type, but with no fields set other than the resourceVersion
-	testCheckResultFunc(t, watch.Bookmark, w, func(object runtime.Object) error {
+	testCheckResultFunc(t, w, func(actualEvent watch.Event) {
+		expectObjectsDeepEqual(t, "incorrect event type", watch.Bookmark, actualEvent.Type)
 		// first, check that we have the correct resource version
-		obj, ok := object.(metav1.Object)
+		obj, ok := actualEvent.Object.(metav1.Object)
 		if !ok {
-			return fmt.Errorf("got %T, not metav1.Object", object)
+			t.Fatalf("got %T, not metav1.Object", actualEvent.Object)
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/storage/testing/watcher_tests.go:554]: t.Errorf

same below

Comment (p0lyn0mial) [reply staging/src/k8s.io/apiserver/pkg/storage/testing/watcher_tests.go:554]: we should stop here, there is no point in checking anything else. I prefer t.Fatal rather than e.Error followed by an empty return statement.

File: staging/src/k8s.io/apiserver/pkg/storage/testing/utils.go (hunk 2 above)
Comment (p0lyn0mial) [reply staging/src/k8s.io/apiserver/pkg/storage/testing/utils.go:107]: the issue is that the output is not readable since this function is used in unit tests it is easier to fail early just like `require.NoError` function does.

File: staging/src/k8s.io/apiserver/pkg/storage/testing/watcher_tests.go (hunk 4 above)
Comment (wojtek-t) [reply staging/src/k8s.io/apiserver/pkg/storage/testing/watcher_tests.go:554]: t.Fatalf means that if we extend the test and will do something after this techCheckResultFunc this will no longer fire in case of error here

File: staging/src/k8s.io/apiserver/pkg/storage/testing/utils.go (hunk 3 above)
Comment (p0lyn0mial) [reply staging/src/k8s.io/apiserver/pkg/storage/testing/utils.go:103]: yeah, naming is hard, the new name is more accurate

File: staging/src/k8s.io/apiserver/pkg/storage/testing/watcher_tests.go (hunk 4 above)
Comment (wojtek-t) [reply staging/src/k8s.io/apiserver/pkg/storage/testing/watcher_tests.go:554]: But I don't feel super strongly about it - I can live with t.Fatalf here.

File: staging/src/k8s.io/apiserver/pkg/storage/testing/utils.go (hunk 3 above)
Comment (p0lyn0mial) [reply staging/src/k8s.io/apiserver/pkg/storage/testing/utils.go:103]: okay, I will rename to `expectNoDiff` :)
